### ISSUE TITLE
Spring 2023 Project Writeup

### ISSUE THREAD

**Issue by JusticeV452** [dhmit/ottoman-sarki#11]
Now that we are at the end of the semester, we need to document work done on the project so that future work can pick up as seemingly as possible. Here is the link to the writeup, which we will all work on and add to the repository once it is complete: [Ottoman Sarki Project Spring 2023 Writeup](https://docs.google.com/document/d/1ISyftfLe68rMrLNrP_YOxb5mVXB_-9pvUUZ0qV1n874/edit?usp=sharing)

It would be helpful to make a note of:
- Scores you transcribed
- Transcription tips/tips for working with Muscore and the musicxml files
- Scores you put in questionable scores and why you put them there 
- Any code you worked on
- Ideas you have for future work
- Anything else you think would be helpful to someone working on the project in the future

**Comment by reeceyang** [comment]
My writeup (copied here in case of any future Google doc issues):


-   Scores you transcribed:
    

	-   CT‐178 Nevâ Bûselik Sâz Semâ’îsi. Tatar’ın
	    
	-   CT‐179 Hisâr Bûselik Peşrevi. Zeki Mehmed Ağa [Usulü Hafîf]
	    
	-   CT‐187 Hisâr Bûselik Sâz Semâ’îsi ‘Osmân Bey
	    
	-   CT‐184 ‘Acem Bûselik Peşrevi. Tosun Ağa [Çenber]
	    
	-   CT‐186 ‘Acem Bûselik Sâz Semâ’îsi. Tosun Ağa
	    
	-   CT‐187 Beyâtî Bûselik Peşrevi. Andon’un [Düyek]
	    
	-   CT‐191 Beyâtî Bûselik Sâz Semâ’îsi. Andon’un
	    
	-   CT‐192 Beyâtî ‘Arabân‐Bûselik Peşrevi. Fahrî Efendi [Fahte]
	    
	-   CT‐194 Beyâti ‘Arabân‐Bûselik Sâz Semâ’îsi. Fahrî Efendi
	    
	-   CT‐195 ‘Arazbâr Bûselik Peşrevi Zekî Mehmed Ağa [Düyek]
    

-   Transcription tips/tips for working with Muscore and the musicxml files
    

	-   Learning to use Musescore keyboard shortcuts helps speed things up
	    
	-   You can set Musescore to infinite vertical scroll mode to avoid having to jump around to different pages while transcribing
	    
	-   Having the score and Musescore side-by-side next to each other (I stacked the windows on top of each other) on your display also helps
    

-   Scores you put in questionable scores and why you put them there
    

	-   The questionable parts in the scores are all marked with text or question marks. Git commit history will have a record of who put the score into the folder.
    

-   Any code you worked on
    

	-   I worked on replace_key_signatures.ipynb. This code looks at the filename of a sarki to find the corresponding maqam and its key signature, then inserts xml representing that key signature into the sarki xml. It does this for every file in the specified folder. `STRKS` stands for string key signature (this was carried over from the old code). `NAME_TO_MSTRKS` maps sets of maqam names to the corresponding key signature’s list of accidentals. None of the maqam names have accents, so we use `remove_accents` on the filenames before attempting to match them up with a maqam.
	    
	-   Music21 accidental handling issue: [https://github.com/dhmit/ottoman-sarki/issues/10](https://github.com/dhmit/ottoman-sarki/issues/10)
	    
	-   find_teslim.ipynb: I tried to determine the maqam of the teslims by seeing how many pitches they had in common. This happens in the last block of find_teslim.ipynb. It makes a set of all pitches in the teslim, a set of all pitches in the maqam, then finds the pitches common to both sets, the pitches that were only in the teslim, and the pitches that were only in the maqam. I used the assumption that the correct maqam maximizes the number of common pitches to come up maqam predictions with respective likelihoods for all sarkis with key signatures.
    

-   Ideas you have for future work
    

	-   See issues ([https://github.com/dhmit/ottoman-sarki/issues](https://github.com/dhmit/ottoman-sarki/issues)) for ideas and tasks for future work
	    
	-   Being able to playback all the pitches in the sarkis correctly would be cool. There are a variety of microtonal tools available on the Internet and looking into using them to play the sarki’s could be an interesting future project.
    

-   Anything else you think would be helpful to someone working on the project in the future
    

	-   Also check out closed issues for previous work and discussion: [https://github.com/dhmit/ottoman-sarki/issues?q=is%3Aissue+is%3Aclosed](https://github.com/dhmit/ottoman-sarki/issues?q=is%3Aissue+is%3Aclosed)
	    

	- Feel free to contact me ([https://github.com/reeceyang](https://github.com/reeceyang)) with any questions or issues
